quantum/action_tapping.c

```diff
@@ -117,6 +117,75 @@ void action_tapping_process(keyrecord_t record) {
     }
 }
 
+/* Some conditionally defined helper macros to keep process_tapping more
+ * readable
+ */
+#    if (defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT)) || defined(TAPPING_TERM_PER_KEY) || defined(PERMISSIVE_HOLD_PER_KEY) || defined(TAPPING_FORCE_HOLD_PER_KEY) || defined(HOLD_ON_OTHER_KEY_PRESS_PER_KEY)
+#        define DEFINE_tapping_keycode uint16_t tapping_keycode = get_record_keycode(&tapping_key, false)
+#        define GET_TAP_TAPPING_TERM() GET_TAPPING_TERM(tapping_keycode, &tapping_key)
+#    else
+#        define DEFINE_tapping_keycode
+#        ifdef DYNAMIC_TAPPING_TERM_ENABLE
+#            define GET_TAP_TAPPING_TERM() g_tapping_term
+#        else
+#            define GET_TAP_TAPPING_TERM() TAPPING_TERM
+#        endif
+#    endif
```

Comment:
```suggestion
```
Your `GET_TAP_TAPPING_TERM` is redundant with `GET_TAPPING_TERM`. `GET_TAPPING_TERM` already does what you want, see https://github.com/qmk/qmk_firmware/blob/8f585153c470b07bb0c529ff49b39ef45f68d37e/quantum/action_tapping.h#L48-L54=

Reply:
The problem is, that GET_TAPPING_TERM requires a keycode argument. But tapping_key may not be declared, so I can't use it unconditionally.